packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+
+```
+                                                    |
+                     +-----------------+            |     +--------------+
+                     |                 |            |     |              |
+                     |  CHAIN-SERVICE  <------------------>  BLOCKCHAIN  |
+                     |                 |            |     |              |
+                     +--------^--------+            |     +--------------+
+                              |                     |
+                              |                     |
+                        +-----v----+                |     +----------+
+                        |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  API Request   |     |          |
+                        |          <-----------------------          |
+                        |          |  API Response  |     |          |
+         +---------+    |          ----------------------->          |
+         |         |    |          |                |     |          |
+         |  STORE  <--->|  WALLET  |  Events        |     |   APP    |
+         |         |    |          ----------------------->          |
+         +---------+    |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  pushMessage   |     |          |
+                        |          <---------------------------+     |
+                        |          |                |     |    |     |
+                        |          |                |     |    |     |
+                        +----------+                |     +----|-----+
+                                                    |          |
+                                                    | +--------|---------+
+                                                    | |  COUNTERPARTY    |
+                                                    | |  WALLETS         |
+                                                      +------------------+
+```
+
+# The responsibilities of a state channel wallet
+
+As in the [Hippocratic oath](https://en.wikipedia.org/wiki/Hippocratic_Oath): _primum non nocere,_ (first, do no harm).
+
+1. Do not lose secrets (private keys)
+2. Do not leak secrets (private keys)
+3. Store relevant states and signatures (Don’t lose any unless they are screened off by kept ones, allow states to be queried)
+4. Do not sign states or blockchain transactions unless the App grants permission
+
+Next, do some active things to protect The App’s interests.
+
+5. With implicit permission, allow an exit on chain without losing funds
+6. Detect and store blockchain challenges (for querying)
+7. Detect and store changes in channel funding (for querying)
```

Comment:
What's the purpose of the (for querying) addition here? I feel like I'd understand 6-7 without it, but adding it makes me confused.
```suggestion
6. Detect and store blockchain challenges
7. Detect and store changes in channel funding
```